llvm/tools/llvm-cgdata/Opts.td

```diff
@@ -0,0 +1,28 @@
+include "llvm/Option/OptParser.td"
+
+class F<string letter, string help> : Flag<["-"], letter>, HelpText<help>;
+
+// General options
+def generic_group : OptionGroup<"Genric Options">, HelpText<"Generic Options">;
+def help : F<"help", "Display this help">, Group<generic_group>;
+def : Flag<["-"], "h">, Alias<help>, HelpText<"Alias for --help">, Group<generic_group>;
+def version : F<"version", "Display the version">, Group<generic_group>;
+def : Flag<["-"], "v">, Alias<version>, HelpText<"Alias for --version">, Group<generic_group>;
+
+// Action options
+def action_group : OptionGroup<"Action">, HelpText<"Action (required)">;
+def show : F<"show", "Show summary of the (indexed) codegen data file.">,
```

Comment:
I think we should consider extending https://github.com/llvm/llvm-project/blob/main/llvm/lib/Option/Option.cpp to support subcommands akin to https://github.com/llvm/llvm-project/blob/main/llvm/lib/Support/CommandLine.cpp.

Reply:
Yes, I agree that it would be beneficial to have subcommand support in `Opt`. However, as discussed in https://github.com/llvm/llvm-project/pull/100077, there seems no explicit support for it yet, and I am not familiar with how to implement this feature. For the time being, I have looked into similar implementations, and tools like `sandcov` group these options as required to mimic subcommand-like behavior, which I'm following -- 
https://github.com/llvm/llvm-project/blob/main/llvm/tools/sancov/Opts.td#L16-L38